stdlib/format.ml

```diff
@@ -442,11 +442,15 @@ let rec advance_left state =
   | Some { size; token; length } ->
     let pending_count = state.pp_right_total - state.pp_left_total in
     if Size.is_known size || pending_count >= state.pp_space_left then begin
-      Queue.take state.pp_queue |> ignore; (* Not empty: we peek into it *)
-      let size = if Size.is_known size then Size.to_int size else pp_infinity in
-      format_pp_token state size token;
-      state.pp_left_total <- length + state.pp_left_total;
-      (advance_left [@tailcall]) state
+      match Queue.take_opt state.pp_queue with
+      | None -> invalid_arg "Format: Unsynchronized access to formatter"
+      | Some _ ->  (* Not empty: we peek into it *)
```

Comment:
I'm not sure what the comment means, because we peeked into it *before*, but it suggests that we are now peeking. If I understand correctly, we have decided to commit, drop the element we saw when peeking, and process it. (If you're not sure what to write, I would maybe just drop the comment.)

Reply:
> I'm not sure what the comment means, because we peeked into it _before_, but it suggests that we are now peeking. If I understand correctly, we have decided to commit, drop the element we saw when peeking, and process it. (If you're not sure what to write, I would maybe just drop the comment.)

This comment was there before and I kept it unchanged. We peek_opt the element and if there is enough room we peek it and print it.